PhysEdJournal.Infrastructure/Commands/AdminCommands/ArchiveStudentCommand.cs

```diff
@@ -66,11 +60,98 @@ ArchiveStudentCommandPayload commandPayload
 
         var totalPoints = CalculateTotalPoints(
             studentFromDb.Visits,
-            studentFromDb.VisitValue,
+            studentFromDb.Group.VisitValue,
             studentFromDb.AdditionalPoints,
             studentFromDb.PointsForStandards
         );
 
+        var visitsToArchive = studentFromDb.VisitsStudentHistory;
+        var pointsToArchive = studentFromDb.PointsStudentHistory;
+        var standardsToArchive = studentFromDb.StandardsStudentHistory;
+
+        if (studentFromDb.HasDebtFromPreviousSemester && totalPoints > REQUIRED_POINT_AMOUNT)
+        {
+            var histories = FindHistoriesToArchive(
+                studentFromDb.ArchivedVisitValue,
+                studentFromDb.PointsStudentHistory,
+                studentFromDb.StandardsStudentHistory,
+                studentFromDb.VisitsStudentHistory
+            );
+
+            visitsToArchive = new List<VisitStudentHistoryEntity>();
+            pointsToArchive = new List<PointsStudentHistoryEntity>();
+            standardsToArchive = new List<StandardsStudentHistoryEntity>();
```

Comment:
Можно сделать так:
```suggestion
            visitsToArchive = studentFromDb.VisitsHistory.Filter(h => histories.Include(h.Id));
            pointsToArchive = studentFromDb.PointsHistory.Filter(h => histories.Include(h.Id));
            standardsToArchive = studentFromDb.StandardsHistory.Filter(h => histories.Include(h.Id));
```
Так получается и тип даже не нужен в этом массиве

Reply:
сомнительно